File: receipt-analyzer-service/src/main/java/pl/madzierski/daniel/app/receipt/model/GetReceiptRevisionsResponse.java
package pl.madzierski.daniel.app.receipt.model;

import pl.madzierski.daniel.app.receipt.revision.ReceiptRevisionEntity;
import pl.madzierski.daniel.app.receipt.scan_resolver.ReceiptResolverStrategyType;

import java.time.LocalDateTime;

public record GetReceiptRevisionsResponse(
        String id,
        ReceiptResolverStrategyType resolver,
        LocalDateTime createdDate,
        String brand,
        Double totalPrice,
        String payingDate,
        String address,
        Boolean isPreferredRevision,
        Boolean isCorrect
) {
    public static GetReceiptRevisionsResponse receiptRevisionMapper(ReceiptRevisionEntity revision) {
        return new GetReceiptRevisionsResponse(
                revision.getId(),
                revision.getResolver(),
                revision.getCreatedDate(),
                revision.getBrand(),
                revision.getTotalPrice(),
                revision.getPayingDate(),
                revision.getAddress(),
                revision.getIsPreferredRevision(),
                revision.getIsCorrect()
        );
    }
}

File: receipt-analyzer-service/src/main/kotlin/pl/madzierski/daniel/app/receipt/model/GetReceiptRevisionsResponse.kt
package pl.madzierski.daniel.app.receipt.model

import pl.madzierski.daniel.app.receipt.revision.ReceiptRevisionEntity
import pl.madzierski.daniel.app.receipt.revision.ScanResolver
import java.time.LocalDateTime

data class GetReceiptRevisionsResponse(
    var id: String?,
    var resolver: ScanResolver?,
    val createdDate: LocalDateTime?,
    val brand: String?,
    val totalPrice: Double?,
    val payingDate: String?,
    val address: String?,
    val isPreferredRevision: Boolean?,
    val isCorrect: Boolean?,

    ) {
    companion object {
        fun receiptRevisionMapper(revision: ReceiptRevisionEntity): GetReceiptRevisionsResponse =
            GetReceiptRevisionsResponse(
                revision.id,
                revision.resolver,
                revision.createdDate,
                revision.brand,
                revision.totalPrice,
                revision.payingDate,
                revision.address,
                revision.isPreferredRevision,
                revision.isCorrect
            )


    }
}